app/controllers/concerns/generic_crud.rb

```diff
@@ -15,47 +9,64 @@ def create
   end
 
   def show
-    render json: send(resource)
+    render json: resource
   end
 
   # TODO: enforce user authorization
   def update
     redirect_to root_path unless valid_user?
-    send(resource).update!(params_attributes)
+    resource.update!(params_attributes)
 
-    render json: send(resource)
+    render json: resource
   end
 
   # TODO: mark user as anonymous instead of deleting it
   # Maybe update the user to -1: anonymous user
   # or use paranoid to retrieve all questions
   def destroy
     redirect_to root_path unless valid_user?
-    send(resource).destroy
+    resource.destroy
 
     render json: { message: 'Delete Successful' }
   end
 
+  private
+
+  def options
+    @options ||= { user: user }
+  end
+
+  # TODO: create a generic getter from params
+  # :user_id, :question_id
+  def user
+    @user ||= User.find(params[:user_id])
+  end
+
+  def valid_user?
+    resource.user_id == user.id
+  end
+
   # returns singularize resource name from url
   # /questions -> "question"
   # /questions/2 -> "question"
   # /questions?page=2&limit=10 -> "question"
-
-  private
-
-  def resource
+  def resource_name
     remove_query_params = request.url.split('?')[0]
     split_url = remove_query_params.split('/')
-    resource_name = if split_url[-1].to_i.zero?
-                      split_url[-1]
-                    else
-                      split_url[-2]
-                    end
+    name = if split_url[-1].to_i.zero?
```

Comment:
try get model name from Controllers name

Reply:
should I create a getter in each controller

`# questions_controller.rb`
```ruby
def resource_name
    Question
end
```